File: ChatApplication/src/main/java/com/web/chatapplication/models/MessageModel.java
package com.web.chatapplication.models;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Document(collection = "MessageModel")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode
public class MessageModel {

    @Id
    private String id;
    private String usernameOfSender;
    private String userId;
    private String chatId;
    private List<String> usersSoftDeletingMessage;
    private boolean softDeleted = false;
    @NonNull
    private String text;
    private LocalDateTime timeOfSendingMessage;
    private List<String> seenUserIdList; // if message has been seen or not we put the id of users here;
}

File: src/app/models/message.model.ts

export class MessageModel {
  id?: string;

  usernameOfSender?: string;
  userId?: string;
  text?: string;
  chatId?: string;
  usersSoftDeletingMessages!: string[];
  softDeleted!: boolean;
  timeOfSendingMessage?: Date;
  seenUserIdList?: string[];
}
